core/metacling/src/TClingCallFunc.cxx

```diff
@@ -1389,384 +1389,85 @@ TClingCallFunc::make_dtor_wrapper(const TClingClassInfo *info)
    return (tcling_callfunc_dtor_Wrapper_t)F;
 }
 
-class ValHolder {
-public:
-   union {
-      long double ldbl;
-      double dbl;
-      float flt;
-      //__uint128_t ui128;
-      //__int128_t i128;
-      unsigned long long ull;
-      long long ll;
-      unsigned long ul;
-      long l;
-      unsigned int ui;
-      int i;
-      unsigned short us;
-      short s;
-      //char32_t c32;
-      //char16_t c16;
-      //unsigned wchar_t uwc; - non-standard
-      wchar_t wc;
-      unsigned char uc;
-      signed char sc;
-      char c;
-      bool b;
-      void *vp;
-   } u;
-};
-
 void TClingCallFunc::exec(void *address, void *ret)
 {
-   SmallVector<ValHolder, 8> vh_ary;
-   SmallVector<void *, 8> vp_ary;
+   const FunctionDecl *FD = GetDecl();
```

Comment:
After the simplification my claim is that we don't need to lock as we have const operations. @pcanal, can you take a look?

Reply:
@pcanal, ping - can you go over the locks - I believe we can lift some of them.

Reply:
`const` operation here are not good enough to know whether the lock is necessary or not because the 'clang database' is not fine grained locked (at least not in our builds) and thus another thread can modify the database (unload decl or add template instantiation decls for example) while this thread is doing `const` operations.